manufacturing-client/src/main.rs

```diff
@@ -937,3 +960,24 @@ impl KeyReference {
         }
     }
 }
+
+fn get_default_network_iface() -> Result<Option<String>, std::io::Error> {
+    let file = File::open("/proc/net/route")?;
+    let reader = BufReader::new(file);
+
+    for line in reader.lines().skip(1) {
+        let line = line.unwrap();
+        let fields: Vec<_> = line.split_whitespace().collect();
+        if fields.is_empty() {
+            continue;
+        }
+        // /proc/net/route this is a system file which gives info about kernel's routing table and
```

Comment:
What about IPv6?

Reply:
Thats a great question!
I will add support to read ipv6 interfaces too.

Reply:
@nullr0ute added support for ipv6 network interfaces, ptal!

Reply:
as discussed ,I will create another PR for ipv6, This PR focusses on ipv4 network interface support.